### PR TITLE
FairingDecoupler: apply force to part, not to its Rigidbody.

### PR DESCRIPTION
**Pull request by fat-lobyte**
Courtesy of @eggrobin:

Route force addition through part mechanism, so that Principia can register and apply the forces.

Fixes fairing deployment in space when using newer Principia versions.

### CODE REVIEW

File: Source/ProceduralFairings/FairingDecoupler.cs [hunk 1]
```diff
@@ -112,8 +112,8 @@ public void FixedUpdate ()
 
                     if (tr)
                     {
-                        part.Rigidbody.AddForce (tr.TransformDirection (forceVector) * Mathf.Lerp (ejectionLowDv, ejectionDv, ejectionPower), ForceMode.VelocityChange);
```
Comment (eggrobin) [inline Source/ProceduralFairings/FairingDecoupler.cs:115]: Removing this `ForceMode.VelocityChange` will change the semantics: with `ForceMode.VelocityChange` we were passing a ∆v.
We would need to turn that into an actual force, `Part.AddForce` has no such option.
F=p′=∆p/∆t=m∆v/∆t.

@dkavolis points out that there is a `Part.AddImpulse` that takes a change in linear momentum, so we can let KSP take care of the ∆t for us, and just pass ∆p=m∆v.

```C#
Vector3d Δv = tr.TransformDirection (forceVector) * Mathf.Lerp (ejectionLowDv, ejectionDv, ejectionPower);
part.AddImpulse(part.mass * Δv);
```

File: Source/ProceduralFairings/FairingDecoupler.cs [hunk 2]
```diff
@@ -112,8 +112,8 @@ public void FixedUpdate ()
...
-                        part.Rigidbody.AddTorque (tr.TransformDirection (torqueVector) * Mathf.Lerp (ejectionLowTorque, ejectionTorque, torqueAmount), ForceMode.VelocityChange);
```
Comment (eggrobin) [inline Source/ProceduralFairings/FairingDecoupler.cs:116]: Here we have to do it all ourselves.
L=Iω,
τ=L′=Iω′=I∆ω/∆t = ∆L/∆t.
For reference, I am using the following lines of Principia code as a guide to interpret `inertiaTensorRotation` and `rotation`:
https://github.com/mockingbirdnest/Principia/blob/66661d78f531791d2d1e7e3a135ecd13fe30a4f6/ksp_plugin_adapter/ksp_plugin_adapter.cs#L1100-L1108
https://github.com/mockingbirdnest/Principia/blob/48f2466671bb206fc0999da38f36c1eb3b6a6f80/ksp_plugin/interface.cpp#L861-L864,
https://github.com/mockingbirdnest/Principia/blob/2bd9be1cc2b8041972bbaf8e48ac57214b5f2af9/ksp_plugin/interface_body.hpp#L532-L536

```C#
Vector3 ΔωWorld = tr.TransformDirection (torqueVector) * Mathf.Lerp (ejectionLowTorque, ejectionTorque, torqueAmount);
Vector3 principalMomentsOfInertia = part.RigidBody.inertiaTensor;
// part.RigidBody.inertiaTensorRotation converts coordinates in the principal axes of
// the part to coordinates in the axes of the part.
// part.RigidBody.rotation converts coordinates in the axes of the part to World
// coordinates.
Quaternion principalAxesToWorld = part.RigidBody.rotation * part.RigidBody.inertiaTensorRotation;
Quaternion worldToPrincipalAxes = Quaternion.Inverse(principalAxesToWorld);
Vector3 ΔωPrincipalAxes = worldToPrincipalAxes * ΔωWorld;
Vector3 ΔLPrincipalAxes = Vector3.Scale(principalMomentsOfInertia, ΔωPrincipalAxes);
Vector3 ΔLWorld = principalAxesToWorld * ΔLprincipalAxes;
part.AddTorque(ΔLWorld / TimeWarp.fixedDeltaTime);
```
@pleroy can you double-check that I got the reference frames right here?

**Review by eggrobin — CHANGES_REQUESTED** [review]
@fat-lobyte please retest in-game (with Principia, and probably without as well for good measure) after making the suggested changes.

File: Source/ProceduralFairings/FairingDecoupler.cs [hunk 3]
```diff
@@ -112,8 +112,22 @@ public void FixedUpdate ()
...
+                        Vector3d Δv = tr.TransformDirection(forceVector) * Mathf.Lerp(ejectionLowDv, ejectionDv, ejectionPower);
+                        part.AddImpulse(part.mass * Δv);
+
+
```
Comment (pleroy) [inline Source/ProceduralFairings/FairingDecoupler.cs]: Maybe drop one if these empty lines?  One line good, two lines bad.

Comment (fat-lobyte) [reply Source/ProceduralFairings/FairingDecoupler.cs]: I agree. Done, fixed in the latest push.

**Comment by pleroy** [comment]
Look good, but then I have no power here.

**Review by eggrobin — APPROVED** [review]
Tested a test build provided by @fat-lobyte, with Principia (a post-Fubini build), with the save
[procfairing-in-orbit.zip](https://github.com/KSP-RO/ProceduralFairings/files/4604071/procfairing-in-orbit.zip) provided by @fat-lobyte.

With https://github.com/KSP-RO/ProceduralFairings/releases/tag/v1.8.1, the fairing gets stuck, Gemini 9A-style.
With the new code, the fairing detaches normally.

**Pull request merged**